x-pack/plugins/monitoring/public/alerts/badge.tsx

```diff
@@ -4,187 +4,140 @@
   const inSetupMode = isInSetupMode(React.useContext(SetupModeContext));
-  const alerts = Object.values(props.alerts).filter((alertItem) => Boolean(alertItem?.rawAlert));
+  const alertsContext = React.useContext(AlertsContext);
+  const alertCount = inSetupMode
+    ? alerts.length
+    : alerts.reduce(
+        (sum, { states }) => sum + states.filter(({ state }) => stateFilter(state)).length,
+        0
+      );
+  const [showByNode, setShowByNode] = React.useState(
+    !inSetupMode && alertCount > MAX_TO_SHOW_BY_CATEGORY
+  );
+
+  React.useEffect(() => {
+    if (inSetupMode && showByNode) {
+      setShowByNode(false);
+    }
+  }, [inSetupMode, showByNode]);
 
-  if (alerts.length === 0) {
+  if (alertCount === 0) {
     return null;
   }
 
-  const badges = [];
-
-  if (inSetupMode) {
-    const button = (
-      <EuiBadge
-        iconType="bell"
-        onClickAriaLabel={numberOfAlertsLabel(alerts.length)}
-        onClick={() => setShowPopover(true)}
-      >
-        {numberOfAlertsLabel(alerts.length)}
-      </EuiBadge>
-    );
-    const panels = [
-      {
-        id: 0,
-        title: i18n.translate('xpack.monitoring.alerts.badge.panelTitle', {
-          defaultMessage: 'Alerts',
-        }),
-        items: alerts.map(({ rawAlert }, index) => {
-          return {
-            name: <EuiText>{rawAlert.name}</EuiText>,
-            panel: index + 1,
-          };
-        }),
-      },
-      ...alerts.map((alertStatus, index) => {
-        return {
-          id: index + 1,
-          title: alertStatus.rawAlert.name,
-          width: 400,
-          content: <AlertPanel alert={alertStatus} />,
-        };
-      }),
-    ];
-
-    badges.push(
-      <EuiPopover
-        id="monitoringAlertMenu"
-        button={button}
-        isOpen={showPopover === true}
-        closePopover={() => setShowPopover(null)}
-        panelPaddingSize="none"
-        anchorPosition="downLeft"
-      >
-        <EuiContextMenu initialPanelId={0} panels={panels} />
-      </EuiPopover>
-    );
-  } else {
-    const byType = {
-      [AlertSeverity.Danger]: [] as AlertInPanel[],
-      [AlertSeverity.Warning]: [] as AlertInPanel[],
-      [AlertSeverity.Success]: [] as AlertInPanel[],
-    };
-
-    for (const alert of alerts) {
-      for (const alertState of alert.states) {
-        if (alertState.firing && stateFilter(alertState.state)) {
-          const state = alertState.state as AlertState;
-          byType[state.ui.severity].push({
-            alertState,
-            alert,
-          });
-        }
+  let groupByType = GROUP_BY_NODE;
+  for (const alert of alerts) {
+    for (const { state } of alert.states) {
+      switch (state.stackProduct) {
+        case ELASTICSEARCH_SYSTEM_ID:
+        case LOGSTASH_SYSTEM_ID:
+          groupByType = GROUP_BY_NODE;
+          break;
+        case KIBANA_SYSTEM_ID:
+        case BEATS_SYSTEM_ID:
+          groupByType = GROUP_BY_INSTANCE;
+          break;
       }
     }
+  }
```

Comment:
We don't need this anymore, since there is no `stackProduct`

Reply:
done